### PR TITLE
Alerting: Only warm alert state cache if execute_alerts=true.

### PR DESCRIPTION
**Pull request by stevesg**
If the Grafana instance is not executing alerts, then Warm()-ing the state
manager is wasteful and could lead to misleading rule status queries, as the
status returned will be always based on the state loaded from the database at
startup, and not the most recent evaluation state.

### CODE REVIEW

File: pkg/services/ngalert/ngalert.go [hunk 1]
```diff
@@ -352,9 +352,15 @@ func (ng *AlertNG) Run(ctx context.Context) error {
...
+	ng.Log.Debug("Starting", "execute_alerts", ng.Cfg.UnifiedAlerting.ExecuteAlerts)
 
-	ng.stateManager.Warm(ctx, ng.store)
+	// If we are not executing alerts, then Warm()-ing the state manager is
+	// wasteful and could lead to misleading rule status queries, as the status
+	// returned will be always based on the state loaded from the database at
+	// startup, and not the most recent evaluation state.
+	if ng.Cfg.UnifiedAlerting.ExecuteAlerts {
+		ng.stateManager.Warm(ctx, ng.store)
+	}
```
Comment (gotjosh) [inline pkg/services/ngalert/ngalert.go:363]: nit: Shoul we move both to the same block below? I think we want the Warming to be sync so it should be outside of the callout to the scheduler routine but can be within the same conditional block.

**Review by gotjosh — APPROVED** [review]
LGTM

I've added the backport label for 9.4x to see if it's feasible that we port it back - I'm happy to take on the backport PR (just assign it to me when you merge this one)

File: pkg/services/ngalert/ngalert.go (hunk 1 above)
Comment (stevesg) [reply pkg/services/ngalert/ngalert.go:363]: Makes sense, should it use `ctx` or `subCtx`? I think `ctx` but that will look inconsistent.

Comment (gotjosh) [reply pkg/services/ngalert/ngalert.go:363]: I think so too `ctx` would be my boat.

**Review by yuri-tceretian — APPROVED** [review]
LGTM

**Review by gotjosh — APPROVED** [review]

**Pull request merged**